### PR TITLE
Add doc 'how to run samples with Spring Boot 3'

### PR DESCRIPTION
**Pull request by moarychan**
## Purpose
Tell users how to run samples with Spring Boot 3, and provide a template doc to update the Spring Boot 3 support in sub-samples.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
N/A

## What to Check
N/A

### CODE REVIEW

File: HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# How to run samples with Spring Boot 3?
+
+The current examples are built based on Spring Cloud Azure 4.x by default. If you want to use Spring Cloud Azure 6.x based on Spring Boot 3, this document will tell you how to do it.
+
+## How to make a sample support running with Spring Boot 3?
```
Comment (chenrujun) [inline HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md:5]: The main reader of this file should be customer, not sample maintainer.

So I suggest to change :
```
this document will tell you how to do it.

## How to make a sample support running with Spring Boot 3?
1. It's required to add a document *HOW_TO_RUN_THIS_SAMPLE_WITH_SPRING_BOOT_3.md* to introduce the differences, it should be put together with the sample's *README.md*. The outline and contents of the document are as follows:
```

into:
```
please refer to `HOW_TO_RUN_THIS_SAMPLE_WITH_SPRING_BOOT_3.md` beside each sample's `README.md`. The `HOW_TO_RUN_THIS_SAMPLE_WITH_SPRING_BOOT_3.md` should have contents like this: 
```

File: HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+1. Update parent pom path
+
+2. Update dependencies(optional)
+
+3. Update Java code(optional)
+
+```
+
+2. Update the parent pom file to *azure-spring-boot-samples-6.x-pom.xml*.
```
Comment (chenrujun) [inline HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md:20]: 1. The order is not right. It should not be `2` here.
2. Not only update the parent file path, the artifact id should be udpated, too.
3. The content of `Update parent pom path` should be fixed. How about writing the update parent here? Ah.. the relative path is not fixed. I think we can add a comment here.

How about moving these contents info a new file named `HOW_TO_RUN_THIS_SAMPLE_WITH_SPRING_BOOT_3_TEMPLATE.md` besides current file?

File: HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+3. It's optional to list the different dependencies that need to be updated if any other dependencies are required.
+
+4. It's optional to list the source code change list to be updated if any code needs to be updated.
+
+## How to know if a sample can run with Spring Boot 3?
+
+If a sample can run in Spring Cloud Azure 6.x, the *README.md* file of the sample will clearly state that it supports running with Spring Boot 3, and there will be *HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md* file to guide.
```
Comment (chenrujun) [inline HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md:28]: `If a sample can run in Spring Cloud Azure 6.x, the *README.md* file of the sample will clearly state that`:

->
`If a sample can run in Spring Cloud Azure 6.x, the *README.md* file of the sample will clearly like this:`
```
Current sample is using Spring Cloud Azure 4.x (which is compatible with Spring Boot 2.x). If you want the sample about Spring Cloud Azure 6.x (which is compatible with Spring Boot 3.x), please refer to [HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md](./HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md)
```

File: HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# How to run samples with Spring Boot 3?
+
+The current examples are built based on Spring Cloud Azure 4.x by default. If you want to use Spring Cloud Azure 6.x based on Spring Boot 3, this document will tell you how to do it.
```
Comment (chenrujun) [inline HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md:3]: `The current examples are built based on` -> `Samples in this repository is using `

File: README.md [hunk 5]
```diff
@@ -9,3 +9,9 @@
 
 ## Run Samples Based On Spring Native in Spring Boot 2
 Two Maven profiles have been defined in this project to support compiling Spring applications to native executables: `buildpack` and `native`. The `buildpack` profile will use [Buildpacks](https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#getting-started-buildpacks) and the `native` profile will use [Native Build Tools](https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#getting-started-native-build-tools). Please follow the [storage-blob-native sample](spring-native/storage-blob-native) for more details.
+
+
+## How to run samples with Spring Boot 3?
```
Comment (chenrujun) [inline README.md:14]: How about deleting `HOW_TO_RUN_SAMPLES_WITH_SPRING_BOOT_3.md` and paste the contents here?

```
## Samples for Spring Boot 3
1. In each sample's README.md, there is content like this:
> Current sample is using Spring Cloud Azure 4.x (which is compatible with Spring Boot 2.x). If you want the sample about Spring Cloud Azure 6.x (which is compatible with Spring Boot 3.x), please refer to [CONVERT_SAMPLE_TO_USE_SPRING_BOOT_3.md](./CONVERT_SAMPLE_TO_USE_SPRING_BOOT_3.md)
```
If you are adding a new sample, here is the template file about CONVERT_SAMPLE_TO_USE_SPRING_BOOT_3_TEMPLATE.md: [./CONVERT_SAMPLE_TO_USE_SPRING_BOOT_3_TEMPLATE.md](CONVERT_SAMPLE_TO_USE_SPRING_BOOT_3.md)

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -9,3 +9,12 @@
 
 ## Run Samples Based On Spring Native in Spring Boot 2
 Two Maven profiles have been defined in this project to support compiling Spring applications to native executables: `buildpack` and `native`. The `buildpack` profile will use [Buildpacks](https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#getting-started-buildpacks) and the `native` profile will use [Native Build Tools](https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/#getting-started-native-build-tools). Please follow the [storage-blob-native sample](spring-native/storage-blob-native) for more details.
+
+## Samples for Spring Boot 3
```
Comment (fangjian0423) [inline README.md:13]: Should we add a notice to describe branch `feature/spring-boot-3` is useless ?

**Review by fangjian0423 — APPROVED** [review]

File: .github/workflows/mlc_config.json [hunk 7]
```diff
@@ -23,6 +23,9 @@
     },
     {
       "pattern": "^https://www.microsoft.com/edge"
+    },
+    {
+      "pattern": "^https://dev.mysql.com/downloads/shell/"
```
Comment (chenrujun) [inline .github/workflows/mlc_config.json:28]: Why add this one?

File: azure-spring-boot-samples-6.x-pom.xml [hunk 8]
```diff
@@ -57,4 +57,25 @@
     </plugins>
   </build>
 
+  <repositories>
+    <repository>
+      <id>spring-milestones</id>
+      <name>Spring Milestones</name>
+      <url>https://repo.spring.io/milestone</url>
```
Comment (chenrujun) [inline azure-spring-boot-samples-6.x-pom.xml:64]: Why need this?

File: .github/workflows/mlc_config.json (hunk 7 above)
Comment (moarychan) [reply .github/workflows/mlc_config.json:28]: The link checking is not stable due to this URL, then ignore it first.

File: azure-spring-boot-samples-6.x-pom.xml (hunk 8 above)
Comment (moarychan) [reply azure-spring-boot-samples-6.x-pom.xml:64]: Because the stable version 6.0.0-beta.4 is needed, this version is based on a Spring milestone version.

**Pull request merged**